### PR TITLE
[Netatmo] Addresses Issues #2988, #2761 and remove some obsolete Calendar relat…

### PR DESCRIPTION
**Pull request by clinique**
…ed calls

Removed some potential null warnings

Signed-off-by: Gaël L'hopital <glhopital@gmail.com>

### CODE REVIEW

**Comment by martinvw** [comment]
@lolodomo can you do a quick check?

**Comment by martinvw** [comment]
How this adress / resolved https://github.com/openhab/openhab2-addons/issues/2988

And should you remove some of the existing smart code because the representation property is now supported?

**Comment by clinique** [comment]
Well, in fact #2988 was already adressed in the previous PR before merge.

**Comment by martinvw** [comment]
Are you sure, it was reported by @cweitkamp after the merge?

**Comment by clinique** [comment]
@martinvw : yes , it has been reported after merge but this as been modified in the very last step when working on connection stabilization and the code now clearly ensure that we want to access Welcome before setting a hook, that was not the case before.

**Comment by martinvw** [comment]
@clinique thanks for the explanation.

@cweitkamp could you test / validate this?

File: addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/discovery/NetatmoModuleDiscoveryService.java [hunk 1]
```diff
@@ -127,31 +125,21 @@ private void discoverWelcomeHome() {
...
+        if (firmwareVersion != null) {
+            properties.put(Thing.PROPERTY_VENDOR, "Netatmo");
+            properties.put(Thing.PROPERTY_FIRMWARE_VERSION, firmwareVersion);
+            properties.put(Thing.PROPERTY_MODEL_ID, type);
+            properties.put(Thing.PROPERTY_SERIAL_NUMBER, id);
         }
+        addDiscoveredThing(thingUID, properties, name);
     }
 
     private void addDiscoveredThing(ThingUID thingUID, Map<String, Object> properties, String displayLabel) {
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/discovery/NetatmoModuleDiscoveryService.java]: We should add `withRepresentationProperty(EQUIPMENT_ID)` to the `DiscoveryResultBuilder` statement. Shouldn't we?

Comment (clinique) [reply addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/discovery/NetatmoModuleDiscoveryService.java]: My tests shows that it works without.

**Comment by lolodomo** [comment]
I don't know if representation property is required as I remember @clinique implemented an alternative mechanism, but we cannot say that representation property is implemented if withRepresentationProperty is never called.

**Comment by cweitkamp** [comment]
@martinvw See https://github.com/openhab/openhab2-addons/issues/2988#issuecomment-353732377 . I post my results over there.

**Pull request merged**